backend/middlewares/paramHandler.js

```diff
@@ -0,0 +1,69 @@
+const { Quiz } = require("../models/quiz.model");
+const { QuizAttempt } = require("../models/quizAttempt.model");
+
+const quizParamHandler = async (req, res, next, quizId) => {
+  try {
+    const quiz = await Quiz.findById(quizId).select("-__v");
+    if (!quiz) {
+      return res
+        .status(400)
+        .json({ success: false, message: "Quiz not found!" });
+    }
+    req.quiz = quiz;
+    next();
+  } catch (error) {
+    res.status(400).json({
+      success: false,
+      message: "Couldn't retrieve the quiz",
```

Comment:
Couldn't retrieve the quiz sounds like a server issue. If it is then change the status code to 5XX.